src/poetry/console/commands/group_command.py

```diff
@@ -107,3 +109,33 @@ def project_with_activated_groups_only(self) -> ProjectPackage:
         return self.poetry.package.with_dependency_groups(
             list(self.activated_groups), only=True
         )
+
+    def validate_groups(self, group_options: dict[str, set[str]]) -> bool:
+        """
+        Currently issues a warning if it detects that a group is
+        not part of pyproject.toml
+
+        Can be overridden to adapt behavior.
+        """
+        invalid_options = defaultdict(set)
+        for opt, groups in group_options.items():
+            for group in groups:
+                try:
+                    self.poetry.package.dependency_group(group)
+                except ValueError:
+                    invalid_options[opt].add(group)
```

Comment:
@radoering If you on the Poetry team prefer an error, I think we should consider simply making this error from the start. As a warning only, it's not something users can rely on being caught in CI as they have to always read the logs. I'd view it as "revealing bugs" rather than a breaking change IMHO.